### PR TITLE
Fix race condition on map access

### PR DESCRIPTION
**Pull request by abdul-stripe**
We've been getting crashes a couple times a day that look like:

```
fatal error: concurrent map iteration and map write
```

I'm guessing this is probably the cause, considering this is the only loop that doesn't lock before accessing `jt.jobs`

r? @dug-stripe

### CODE REVIEW

**Comment by dug-stripe** [comment]
can we add tests or deploy this branch (possibly to edge-* services) and see if it resolves the issue?

**Comment by abdul-stripe** [comment]
altho i'm not sure that this the cause of the crash, it's definitely its own bug. 

i'll do a quick deploy on edge-* for sanity. but otherwise i think it's good to merge

File: jobtracker.go [hunk 1]
```diff
@@ -109,6 +109,7 @@ func (jt *jobTracker) runningJobLoop() {
 			continue
 		}
 
+		jt.jobsLock.Lock()
```
Comment (tchow-stripe) [inline jobtracker.go]: this introduces a nested lock so as discussed offline plz test

**Comment by tchow-stripe** [comment]
lgtm

**Comment by tchow-stripe** [comment]
lgtm
